### DISCUSSION TITLE
Add istio_service_name field

### DISCUSSION THREAD

**Discussion by Xunzhuo** [kiali/kiali#4685] [General]
In external_services.istio: there is a field `url_service_version`, by default it is `http://istiod:15014/version`
This field is usually made of `istiod_svc_name` and `istiod_pod_monitoring_port`.
In our scenario, istio service name is like `istiod-1-10-3`,when kiali interacts with istiod service. 
The default service name provided by kiali is not OK with it. So how about adding a field istio_service_name?

If url_service_version is default, and istiod_svc_name is configured, we compute url_service_version with `istiod_svc_name` and `istiod_pod_monitoring_port`, and if url_service_version is configured, we use url_service_version directly.

I am not sure if there is other places interating with istiod service besides getting istio version. If it does, we should add a custom field of istio_service_name more reasonable.

**Comment by jmazzitelli** [comment]
It sounds like Kiali just needs to determine the default `url_service_version` value correctly in all cases. [Docs](https://github.com/kiali/kiali-operator/blob/master/crd-docs/crd/kiali.io_kialis.yaml#L732-L733) say:

`url_service_version: The Istio service used to determine the Istio version. If empty, assumes the URL for the well-known Istio version endpoint.`

So this "url_service_version" when left unset or empty string (which is the default) is supposed to tell Kiali to find the endpoint, if it can.

I wonder if Kiali can just easily determine what the service name is without forcing the user to tell it via yet another config option? We do have [istiod_deployment_name](https://github.com/kiali/kiali-operator/blob/master/crd-docs/crd/kiali.io_kialis.yaml#L723) - I wonder if we can use that to make some k8s api calls to determine its service? I'm just trying to think of something to avoid adding yet another istio config setting the user has to set in the kiali config - the list keeps growing and is getting almost unmanageable now.

**Comment by Xunzhuo** [comment]
So I want to make sure if kiali interacts with istiod service just for getting istio version, if it is, it seems like we can just configure url_service_version

**Comment by Xunzhuo** [comment]
For the optimization purpose, we can consider fetch some infos from Cluster API instead of static configuration. For example, we can get the relationship between istio service and istio deployment by selector and get the relationships between configmap or sidecar_injector configmap and Deployment by volumes 

After optimization, we can just configure istio deployment name
